app/src/main/java/com/twilio/conversations/app/common/extensions/TwilioExtensions.kt

```diff
@@ -176,305 +84,49 @@ suspend fun Conversation.getUnreadMessageCount(): Long? = suspendCancellableCoro
-
-        override fun onSuccess(message: Message) = continuation.resume(message)
-
-        override fun onError(errorInfo: ErrorInfo) = continuation.resumeWithException(ConversationsException(errorInfo))
-    })
-}
-
-suspend fun Conversation.advanceLastReadMessageIndex(index: Long): Long = suspendCoroutine { continuation ->
-    advanceLastReadMessageIndex(index, object : CallbackListener<Long> {
-
-        override fun onSuccess(index: Long) = continuation.resume(index)
-
-        override fun onError(errorInfo: ErrorInfo) = continuation.resumeWithException(ConversationsException(errorInfo))
-    })
-}
-
-suspend fun Conversation.getMessageByIndex(index: Long): Message = suspendCoroutine { continuation ->
-    getMessageByIndex(index, object : CallbackListener<Message> {
-
-        override fun onSuccess(message: Message) = continuation.resume(message)
-
-        override fun onError(errorInfo: ErrorInfo) = continuation.resumeWithException(ConversationsException(errorInfo))
-    })
-}
-
-suspend fun Conversation.removeMessage(message: Message): Unit = suspendCoroutine { continuation ->
-    removeMessage(message, object : StatusListener {
-
-        override fun onSuccess() = continuation.resume(Unit)
-
-        override fun onError(errorInfo: ErrorInfo) = continuation.resumeWithException(ConversationsException(errorInfo))
-    })
-}
-
-suspend fun Message.setAttributes(attributes: Attributes): Unit = suspendCoroutine { continuation ->
-    setAttributes(attributes, object : StatusListener {
-
-        override fun onSuccess() = continuation.resume(Unit)
-
-        override fun onError(errorInfo: ErrorInfo) = continuation.resumeWithException(ConversationsException(errorInfo))
+        override fun onError(errorInfo: ErrorInfo) = continuation.resumeWithException(TwilioException(errorInfo))
     })
 }
 
-suspend fun Conversation.join(): Unit = suspendCoroutine { continuation ->
-    join(object : StatusListener {
-        override fun onSuccess() = continuation.resume(Unit)
-
-        override fun onError(errorInfo: ErrorInfo) = continuation.resumeWithException(ConversationsException(errorInfo))
-    })
-}
-
-suspend fun Conversation.muteConversation(): Unit = suspendCoroutine { continuation ->
-    setNotificationLevel(Conversation.NotificationLevel.MUTED, object : StatusListener {
-        override fun onSuccess() = continuation.resume(Unit)
-
-        override fun onError(errorInfo: ErrorInfo) = continuation.resumeWithException(ConversationsException(errorInfo))
-    })
-}
-
-suspend fun Conversation.unmuteConversation(): Unit = suspendCoroutine { continuation ->
-    setNotificationLevel(Conversation.NotificationLevel.DEFAULT, object : StatusListener {
-        override fun onSuccess() = continuation.resume(Unit)
-
-        override fun onError(errorInfo: ErrorInfo) = continuation.resumeWithException(ConversationsException(errorInfo))
-    })
-}
+suspend fun Conversation.waitForSynchronization(): Conversation {
```

Comment:
`waitForSynchronization` is actually available from SDK, but it doesn't return anything there. 
@rusmonster Should I update the SDK method to return `Conversation` instance?

Reply:
No, it's still usable in chains like `apply { waitForSynchronization() }`

Reply:
Ahh, my bad, totally forgot about kotlin utilities.

Reply:
Applied the suggestion